### DISCUSSION TITLE
rtx command is stuck

### DISCUSSION THREAD

**Discussion by uvlad7** [jdx/mise#3223] [General]
I use mise since it's been rtx, and I didn't reinstall ruby after the migration, just linked old rubies into mise. A lot of tools were expecting rtx, so, as far as I remember, I added `alias rtx='mise'` into my '.bashrc'. A few days ago I updated mise - it was a long time since I updated anything from linuxbrew - and I noticed that bundle commands are often stuck in the end. I found it's because of this
```
File: /home/vladimir/.local/share/mise/installs/ruby/2.3.8/lib/rubygems_plugin/rubygems_plugin.rb
12:     `rtx reshim`
13: 
```
and that the alias is gone and that `rtx` link
```
$ realpath /home/linuxbrew/.linuxbrew/bin/rtx
/home/linuxbrew/.linuxbrew/Cellar/mise/2024.11.29/bin/mise
```
is created. And any command, like `rtx --version`, is stuck.
I replaced it with
```
$ cat /home/linuxbrew/.linuxbrew/bin/rtx                                                                                                                                                                                     
#!/usr/bin/env sh

mise "$@"
```
It works now. But I'm wondering, did mise removed the alias?

**Comment by Marukome0743** [comment]
This is caused by the Ruby installation retaining the old `rubygems_plugin.rb` that was generated before the project was renamed.

The shell alias:

```sh
alias rtx='mise'
```

is only available to the interactive shell, so it is not used when RubyGems executes ``rtx reshim`` from the plugin.

Current mise installations generate this plugin using:

```ruby
`mise reshim`
```

For an existing linked Ruby installation, the permanent fix is therefore either:

- replace ``rtx reshim`` with ``mise reshim`` in its `rubygems_plugin.rb`, or
- reinstall that Ruby version through mise so the current plugin is generated.

Your wrapper script also works as a transitional workaround because it invokes the executable with the actual `mise` name.

I verified that current mise v2026.8.0 no longer hangs indefinitely when invoked through an `rtx` symlink, but it treats `rtx` as a tool shim and exits with:

```text
rtx is not a valid shim
```

Restoring `rtx` executable-name compatibility was proposed in [PR #11641](https://github.com/jdx/mise/pull/11641), but the maintainer declined it as no longer needed. Therefore, existing scripts and old generated plugins should be updated to invoke `mise` directly rather than relying on an `rtx` symlink.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
